src/tox_current_env/hooks4.py

```diff
@@ -146,7 +148,19 @@ def _get_python(self, base_python):
         )
 
     def create_python_env(self):
-        return None
+        # Fake Python environment just to make sure all possible
+        # commands like python or python3 works.
+        self.tempdir = tempfile.TemporaryDirectory()
+        for suffix in (
+            "",
+            f"{sys.version_info.major}",
+            f"{sys.version_info.major}.{sys.version_info.minor}",
+        ):
+            os.symlink(sys.executable, Path(self.tempdir.name) / f"python{suffix}")
+        os.environ["PATH"] = ":".join((os.environ["PATH"], str(self.tempdir.name)))
```

Comment:
Do we need to change the actual `os.environ`? Is there no tox internal environ/path to change?

Reply:
There is an internal one but to change it it'd IMO need to override some more classes in the hierarchy and create our own executor which would change the path before it creates its instance. Do you think it's worth it?

Reply:
I've accomplished this without altering os.environ. See the last commit. The problem, however, is that the new PATH is not propagated to all new processes so tox runs the correct Python (via the symlink in the temp directory) but if I have another python subprocesses in my tests, those will not see the altered PATH. Therefore I'd keep the old behavior and restore the original PATH in the _teardown method. What do you think?

Reply:
I think that propagating that path to all-new subprocesses is a must, but setting it vis os.environ won't propagate it either, will it? It is a matter of how is the inner subprocess invoked. Or what am I not getting? How does `request.env` work? Could you give me an example where the inner subprocess does not respect `request.env` but respects the `os.enviorn` value set from our code?

Reply:
And if we indeed end up manipulating `os.environ` I think we should maximally limit the scope of that by e.g. using a context manager in `<ExecutorClass>.call()` implementation rather than setting it globally when we build the instance.

Reply:
Manipulating the os.environ is definitely not right, for one would break parallel mode.

Reply:
My bad, it was a problem in my testing environment. The current way propagates the PATH to the subprocesses correctly.

> Manipulating the os.environ is definitely not right, for one would break parallel mode.

Our plugin causes tests to run only in one environment. @hroncok why should we care about parallel mode?

Reply:
If you only intend to use it for Fedora sure. But perhaps other people might want to use this for different use cases to use the host Python interpreter where  this might be  required.  It really depends on how you want to scope this project.